play/src/front/Components/Modal/ObjectDetails.svelte

```diff
@@ -0,0 +1,245 @@
+<script lang="ts">
+    import { onDestroy, onMount } from "svelte";
+    import { Unsubscriber, writable } from "svelte/store";
+    import {
+        AreaDataPropertiesKeys,
+        EntityDataPropertiesKeys,
+        OpenWebsiteTypePropertiesKeys,
+    } from "@workadventure/map-editor";
+    import { fly } from "svelte/transition";
+    import { mapEditorModeStore, mapExplorationObjectSelectedStore } from "../../Stores/MapEditorStore";
+    import { Entity } from "../../Phaser/ECS/Entity";
+    import { AreaPreview } from "../../Phaser/Components/MapEditor/AreaPreview";
+    import AreaToolImg from "../images/icon-tool-area.png";
+    import { gameManager } from "../../Phaser/Game/GameManager";
+    import ListAddPropertyButton from "../MapEditor/PropertyEditor/ListAddPropertyButton.svelte";
+    import LL from "../../../i18n/i18n-svelte";
+    import { analyticsClient } from "../../Administration/AnalyticsClient";
+
+    // Create type for component AddPropertyButton
+    type AddPropertyButtonType = {
+        property: AreaDataPropertiesKeys | EntityDataPropertiesKeys;
+        subProperty?: OpenWebsiteTypePropertiesKeys;
+    };
+
+    let iconProperties = writable<Map<string, AddPropertyButtonType>>(new Map());
```

Comment:
I believe you can get completely rid of the iconProperties map and get rid of the AddPropertyButtonType by instead creating a map mapping a string to a Svelte component.

See:

https://svelte.dev/docs/special-elements#svelte-component